src/ParkChecklist/ParkChecklist.js

```diff
@@ -7,222 +7,226 @@ import page from "../resources/page.png";
-            const response = await axios.get(`${apiLink}/${park_code}`);
-            const selectedPark = response.data.data;
+  const fetchData = async () => {
+    try {
+      const response = await axios.get(apiLink);
+      setParks(response.data.data);
+      setLoading(false);
+    } catch (error) {
+      console.error("Error fetching data:", error);
+      setLoading(true);
+    }
+  };
 
-            if (selectedPark) {
-                setParkCode(selectedPark.attributes.park_code);
-                setFoundPark(selectedPark);
-            } else {
-                console.error(`Park with id ${park_code} not found.`);
-            }
-            console.log(selectedPark.attributes.park_code)
-            navigate(`/Parks/${selectedState}/${park_code}`);
-        } catch (error) {
-            console.error("Error fetching park details:", error);
-        }
-    };
+  useEffect(() => {
+    fetchData();
+  }, [apiLink, selectedState, showResults, currentPage, navigate]);
 
-    const fetchData = async () => {
-        try {
-            const response = await axios.get(apiLink);
-            setParks(response.data.data);
-            setLoading(false);
-        } catch (error) {
-            console.error("Error fetching data:", error);
-            setLoading(true);
-        }
-    };
+  const handleCheckboxChange = (parkId) => {
+    setCheckedItems((prevCheckedItems) => {
+      if (prevCheckedItems.includes(parkId)) {
+        return prevCheckedItems.filter((id) => id !== parkId);
+      } else {
+        return [...prevCheckedItems, parkId];
+      }
+    });
+  };
 
-    useEffect(() => {
-        fetchData();
-    }, [apiLink, selectedState, showResults, currentPage, navigate]);
+  const fetchParkDetails = async (park_code) => {
+    try {
+      const response = await axios.get(`${apiLink}/${park_code}`);
+      const selectedPark = response.data.data;
 
-    const handleCheckboxChange = (parkId) => {
-        setCheckedItems((prevCheckedItems) => {
-            if (prevCheckedItems.includes(parkId)) {
-                return prevCheckedItems.filter((id) => id !== parkId);
-            } else {
-                return [...prevCheckedItems, parkId];
-            }
-        });
-    };
+      if (selectedPark) {
+        setParkCode(selectedPark.attributes.park_code);
+        setFoundPark(selectedPark);
+      } else {
+        console.error(`Park with id ${park_code} not found.`);
+      }
+      navigate(`/Parks/${selectedState}/${park_code}`);
+    } catch (error) {
+      console.error("Error fetching park details:", error);
+    } finally {
+    }
+  };
 
-    const handleStateChange = (event) => {
-        setSelectedState(event.target.value);
-    };
+  const handleStateChange = (event) => {
+    setSelectedState(event.target.value);
+  };
 
-    const handleGoClick = () => {
-        setShowResults(true);
-        navigate(`/Parks/${selectedState}`);
-    };
+  const handleGoClick = () => {
+    setShowResults(true);
```

Comment:
was this just a spacing thing? if you did it manually you can instead right click on vs code and click format document to make all the spacing fix itself

Reply:
I didn't realize that. The main thing I was messing with here was trying to make a loading screen in-between the parks and park details pages. I did not successfully get that implemented so I took out the modified code and tried to put the original code back. That must be why the formatting is a little off.